lib/pact/command/cont_command.ex

```diff
@@ -0,0 +1,330 @@
+    cmd_request
+    |> add_keypair(keypair)
+    |> add_keypairs(keypairs)
+  end
+
+  def add_keypairs(%__MODULE__{}, _keypairs), do: {:error, [keypairs: :not_a_list]}
+  def add_keypairs({:error, reason}, _keypairs), do: {:error, reason}
+
+  @impl true
+  def add_signer(%__MODULE__{signers: nil} = cmd_request, %Signer{} = signer),
+    do: %{cmd_request | signers: SignersList.new([signer])}
+
+  def add_signer(%__MODULE__{signers: signer_list} = cmd_request, %Signer{} = signer) do
+    %SignersList{signers: signers} = signer_list
+    %{cmd_request | signers: SignersList.new(signers ++ [signer])}
+  end
+
+  def add_signer(%__MODULE__{}, _signer), do: {:error, [signer: :invalid]}
+  def add_signer({:error, reason}, _signer), do: {:error, reason}
+
+  @impl true
+  def add_signers(%__MODULE__{} = cmd_request, %SignersList{} = list),
+    do: %{cmd_request | signers: list}
+
+  def add_signers(%__MODULE__{} = cmd_request, signers) do
+    case SignersList.new(signers) do
+      %SignersList{} = signers -> %{cmd_request | signers: signers}
+      {:error, reason} -> {:error, [signers: :invalid] ++ reason}
+    end
+  end
+
+  def add_signers({:error, reason}, _signers), do: {:error, reason}
+
+  @impl true
+  def set_pact_tx_hash(%__MODULE__{} = cmd_request, pact_tx_hash) when is_binary(pact_tx_hash),
+    do: %{cmd_request | pact_tx_hash: pact_tx_hash}
+
+  def set_pact_tx_hash(%__MODULE__{}, _pact_tx_hash), do: {:error, [pact_tx_hash: :not_a_string]}
+  def set_pact_tx_hash({:error, reason}, _pact_tx_hash), do: {:error, reason}
+
+  @impl true
+  def set_step(%__MODULE__{} = cmd_request, step) when is_integer(step),
+    do: %{cmd_request | step: step}
+
+  def set_step(%__MODULE__{}, _step), do: {:error, [step: :not_an_integer]}
+  def set_step({:error, reason}, _step), do: {:error, reason}
+
+  @impl true
+  def set_proof(%__MODULE__{} = cmd_request, proof) when is_binary(proof),
+    do: %{cmd_request | proof: proof}
+
+  def set_proof(%__MODULE__{}, _proof), do: {:error, [proof: :not_a_string]}
+  def set_proof({:error, reason}, _proof), do: {:error, reason}
+
+  @impl true
+
+  def set_rollback(%__MODULE__{} = cmd_request, rollback) when is_boolean(rollback),
+    do: %{cmd_request | rollback: rollback}
+
+  def set_rollback(%__MODULE__{}, _rollback), do: {:error, [rollback: :not_a_boolean]}
+  def set_rollback({:error, reason}, _rollback), do: {:error, reason}
+
+  @impl true
+  def build(
+        %__MODULE__{
+          keypairs: keypairs
+        } = cmd_request
+      ) do
+    with {:ok, payload} <- create_payload(cmd_request),
+         {:ok, cmd} <- command_to_json_string(payload, cmd_request),
+         {:ok, sig_commands} <- sign_commands([], cmd, keypairs),
+         {:ok, hash} <- get_unique_hash(sig_commands),
+         {:ok, signatures} <- build_signatures(sig_commands, []),
+         {:ok, command} <- create_command(hash, signatures, cmd) do
+      %Command{} = command
```

Comment:
@EdwinGuayacan @Odraxs IMO, this is not necessary considering that previously `create_command/3` returns a tuple with `{:ok, command}`.